tests/server/dataSource-tests.js

```diff
@@ -0,0 +1,565 @@
+
+sjrk.test.storyTelling.server.dataSource.mockRecords.tagsResponse = {
+    "total_rows": 8,
+    "offset": 0,
+    "rows": [{
+        "key": "Example",
+        "id": "publishedStory",
+        "value": "Sample Story 1"
+    }, {
+        "key": "Example",
+        "id": "storyByOtherAuthor",
+        "value": "Sample Story 2"
+    }, {
+        "key": "Example",
+        "id": "storyExample",
+        "value": "The Story Builder how-to"
+    }, {
+        "key": "Example",
+        "id": "unpublishedStory",
+        "value": "Sample Story 3"
+    }, {
+        "key": "Help",
+        "id": "storyExample",
+        "value": "The Story Builder how-to"
+    }, {
+        "key": "How-to",
+        "id": "storyExample",
+        "value": "The Story Builder how-to"
+    }, {
+        "key": "Sample",
+        "id": "storyByOtherAuthor",
+        "value": "Sample Story 2"
+    }, {
+        "key": "Test",
+        "id": "publishedStory",
+        "value": "Sample Story 1"
+    }]
+};
+
+/**
+ * Converts an object of dbDocuments to a the expected response object from a call to the stories view.
+ *
+ * @param {Object} records - a set of dbDocuments to convert to the expected stories view response.
+ * @return {Object} - a couchDB style view response.
+ */
+sjrk.test.storyTelling.server.dataSource.recordsToStoryViewResponse = function (records) {
+    var filter = ["title", "author", "tags", "content"];
+
+    var response = {
+        offset: 0,
+        rows: []
+    };
+
+    fluid.each(records, function (record, id) {
+        var story = fluid.copy(record.value);
+
+        if (story.published) {
+            story = fluid.filterKeys(story, filter);
+
+            response.rows.push({
+                "key": id,
+                "id": id,
+                "value": story
+            });
+        }
+    });
+    response.total_rows = response.rows.length;
+
+    return response;
+};
+
+fluid.defaults("sjrk.test.storyTelling.server.dataSource.testEnvironment", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    components: {
+        testDB: {
+            type: "sjrk.test.storyTelling.server.mockDatabase",
+            createOnEvent: "{datasourceTester}.events.onTestCaseStart",
+            options: {
+                components: {
+                    authorsDBConfig: {
+                        options: {
+                            dbDocuments: sjrk.test.storyTelling.server.dataSource.mockRecords.authors
+                        }
+                    },
+                    storiesDBConfig: {
+                        options: {
+                            dbDocuments: sjrk.test.storyTelling.server.dataSource.mockRecords.stories
+                        }
+                    }
+                }
+            }
+        },
+        // a DataSource to get a list of stories
+        viewDataSource: {
+            type: "sjrk.storyTelling.server.dataSource.couch.view",
+            options: {
+                host: "http://localhost:6789"
```

Comment:
Perhaps a distributeOptions could be used to inject this host configuration into each test dataSource. This is fine either way, though :)